deals/deals.go

```diff
@@ -225,6 +234,195 @@ func (m *Module) Watch(ctx context.Context, proposals []cid.Cid) (<-chan DealInf
 	return ch, nil
 }
 
+// FinalDealRecords returns a list of all finalized storage deals.
+func (m *Module) FinalDealRecords() ([]DealRecord, error) {
+	return m.store.getFinalDeals()
+}
+
+// PendingDealRecords returns a list of all pending storage deals.
+func (m *Module) PendingDealRecords() ([]DealRecord, error) {
+	return m.store.getPendingDeals()
+}
+
+// AllDealRecords returns a list of all finalized and pending deals.
+func (m *Module) AllDealRecords() ([]DealRecord, error) {
+	ret, err := m.store.getFinalDeals()
+	if err != nil {
+		return nil, err
+	}
+	pending, err := m.store.getPendingDeals()
+	if err != nil {
+		return nil, err
+	}
+	ret = append(ret, pending...)
+	sort.Slice(ret, func(i, j int) bool {
+		if ret[i].Pending || ret[j].Pending {
+			return ret[i].Time < ret[j].Time
+		}
+		if ret[i].DealInfo.ActivationEpoch < ret[j].DealInfo.ActivationEpoch {
+			return true
+		}
+		if ret[i].DealInfo.ActivationEpoch > ret[j].DealInfo.ActivationEpoch {
+			return false
+		}
+		return ret[i].Time < ret[j].Time
+	})
+	return ret, nil
+}
+
+// RetrievalRecords returns a list of all retrievals.
+func (m *Module) RetrievalRecords() ([]RetrievalRecord, error) {
+	return m.store.getRetrievals()
+}
+
+// Close closes the deals Module.
+func (m *Module) Close() error {
+	return m.store.close()
+}
+
+func (m *Module) initPendingDeals() {
+	pendingDeals, err := m.store.getPendingDeals()
+	if err != nil {
+		log.Errorf("getting pending deals: %v", err)
+		return
+	}
+	for _, dr := range pendingDeals {
+		remaining := time.Until(time.Unix(dr.Time, 0).Add(dealTimeout))
+		if remaining <= 0 {
+			go m.finalizePendingDeal(dr)
+		} else {
+			go m.eventuallyFinalizeDeal(dr, remaining)
+		}
```

Comment:
On a `Close()` call, these goroutines won't we waited. 
I think that's OK since things will spin up again re-evaluating all the situation again. 
Mentioning this just in case I may be missing something important to consider.

Reply:
Yea, that is fine because `Close()` could have to wait a loooong time. Correct, the state of things will get picked up where it left off.